app/main.js

```diff
@@ -167,17 +174,84 @@ function createWindow () {
     });
   }
 
+  win.removeAllListeners();
+
+  win.on('close', function(event) {
+    if (minimize) {
+      event.preventDefault();
+      win.hide();
+    }
+  })
+
   win.on('closed', () => {
     win = null
   })
 
+  win.on("hide", () => {
+    if (minimize) updateTray();
+  });
+
+  win.on("show", () => {
+    if (minimize) updateTray();
+  });
+
+  win.on("blur", () => {
+    globalShortcut.unregister('Alt+F4');
+  });
+
+  win.on("focus", () => {
+    globalShortcut.register('Alt+F4', () => safeQuit());
+    if (minimize) updateTray();
+    win.webContents.send('window-is-focussed', null);
+  });
+
   // Menu bar
   win.setAutoHideMenuBar(true);
   win.setMenuBarVisibility(isDebug);
   setMenu();
 
 };
 
+function createTray () {
```

Comment:
Can you add a behavior that double clicking the tray icon also triggers the show behavior?

Reply:
Done.